### DISCUSSION TITLE
config in args - am I doing it right?

### DISCUSSION THREAD

**Discussion by cprn** [nvimtools/none-ls.nvim#182] [Q&A] [answered]
It works fine, but I'm not sure I'm doing it in the right place:
```lua
null_ls.builtins.formatting.shfmt.with({
    args = {"-i", "4", "-ci", "--filename", "$FILENAME"}
}),
```

What if default `args` value changes to something different from `{"-filename", "$FILENAME"}`? Then my config will be broken on update. Is there a dedicated setting for (so to speak) “userspace” arguments (didn't find any in documentation)? Or can I somehow “merge” with the default `args` list instead?

I tried this, and it works, but IDK how safe is the assumption `_opts` won't change:
```lua
null_ls.builtins.formatting.shfmt.with({
    args = {"-i", "4", "-ci", table.unpack(
        null_ls.builtins.formatting.shfmt._opts.args
    )}
}),
```

**Comment by mochaaP** [comment, marked as answer]
You can do this with `extra_args`:

https://github.com/nvimtools/none-ls.nvim/blob/main/doc/BUILTIN_CONFIG.md#Arguments

**Comment by cprn** [comment]
I knew there must be a better way! Although, I even read that page and somehow missed it. Thank you very much!
